src/EditorFeatures/TestUtilities/StubSettingsManagerHost.cs

```diff
@@ -0,0 +1,183 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.ComponentModel;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.PlatformUI;
+using Microsoft.VisualStudio.Settings;
+using Microsoft.VisualStudio.Settings.Telemetry;
+
+namespace Microsoft.CodeAnalysis.Editor.UnitTests;
+
+internal sealed class StubSettingsManagerHost : ISettingsManagerHost5
```

Comment:
If this and the StubLocalRegistry only has to exist because we are now pulling in the VisualStudioOptionsPersister stuff, let's just exclude that from unit tests. It won't ever do anything sane, and just eliminates a bunch of unnecessary mocks.

Reply:
➡️ Based on the fact that this can be reasonably implemented as in-memory string storage, and the factory provides a full implementation of the settings manager service from it, it seems reasonable to keep this as a way to have more accurate overall testing. I might see it differently if we had to implement `ISettingsManager` directly.